src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -569,6 +576,93 @@ def resume_privacy_request(
     return privacy_request
 
 
+def validate_manual_input(
+    manual_rows: List[Row], collection: CollectionAddress, db: Session
+) -> None:
+    """Validate manually-added data for a collection.
+
+    The specified collection must exist and all fields must be previously defined.
+    """
+    datasets = DatasetConfig.all(db=db)
+    dataset_graphs = [dataset_config.get_graph() for dataset_config in datasets]
+    dataset_graph = DatasetGraph(*dataset_graphs)
+
+    node: Optional[Node] = dataset_graph.nodes.get(collection)
+    if not node:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Cannot save manual rows. No collection in graph with name: '{collection.value}'.",
+        )
+
+    for row in manual_rows:
+        for field_name in row:
+            if not dataset_graph.nodes[collection].contains_field(
+                lambda f: f.name == field_name  # pylint: disable=W0640
+            ):
+                raise HTTPException(
+                    status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f"Cannot save manual rows. No '{field_name}' field defined on the '{collection.value}' collection.",
+                )
+
+
+@router.post(
+    PRIVACY_REQUEST_MANUAL_INPUT,
+    status_code=HTTP_200_OK,
+    response_model=PrivacyRequestResponse,
+    dependencies=[
+        Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+    ],
+)
+def resume_with_manual_input(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    cache: FidesopsRedis = Depends(deps.get_cache),
+    manual_rows: List[Optional[Row]],
```

Comment:
> we could surface it on any API endpoint that needs this for the frontend

This will be necessary for the UI. The user will need to know:

- Why is the graph paused?
- Where am I looking for data?
- What fields are we looking for?
- Which identity does this pertain to? (cc @NevilleS as this is another place we'll need to reveal identity data, with potentially more significant impact here as the data required for one datastore might be found inside the customer's internal systems vs. passed in from the requester themselves)
- Where do I submit the data to once found?

Reply:
Yes, exposing this information isn't included in this PR because it's getting long and there were unknowns around what we should expose and where. This is a good starting list; I think we should reticket.

Reply:
We also don't necessarily need the identity data, we need whatever data was passed into that node.  It could have been identity data, but it could also be any other type of data that the user has configured as being an upstream dependency.

Reply:
Followup added here, https://github.com/ethyca/fidesops/issues/570